genrl/bandit/agents/cb_agents/linpos.py

```diff
@@ -0,0 +1,152 @@
+        a0 (float, optional): Inverse gamma prior for noise. Defaults to 6.0.
+        b0 (float, optional): Inverse gamma prior for noise. Defaults to 6.0.
+        device (str): Device to use for tensor operations.
+            "cpu" for cpu or "cuda" for cuda. Defaults to "cpu".
+    """
+
+    def __init__(self, bandit: DataBasedBandit, **kwargs):
+        super(LinearPosteriorAgent, self).__init__(bandit, kwargs.get("device", "cpu"))
+
+        self.init_pulls = kwargs.get("init_pulls", 3)
+        self.lambda_prior = kwargs.get("lambda_prior", 0.25)
+        self.a0 = kwargs.get("a0", 6.0)
+        self.b0 = kwargs.get("b0", 6.0)
+        self.mu = torch.zeros(
+            size=(self.n_actions, self.context_dim + 1),
+            device=self.device,
+            dtype=torch.float,
+        )
+        self.cov = torch.stack(
+            [
+                (1.0 / self.lambda_prior)
+                * torch.eye(self.context_dim + 1, device=self.device, dtype=torch.float)
+                for _ in range(self.n_actions)
+            ]
+        )
+        self.inv_cov = torch.stack(
+            [
+                self.lambda_prior
+                * torch.eye(self.context_dim + 1, device=self.device, dtype=torch.float)
+                for _ in range(self.n_actions)
+            ]
+        )
+        self.a = self.a0 * torch.ones(
+            self.n_actions, device=self.device, dtype=torch.float
+        )
+        self.b = self.b0 * torch.ones(
+            self.n_actions, device=self.device, dtype=torch.float
+        )
+        self.db = TransitionDB(self.device)
+        self.t = 0
+        self.update_count = 0
+
+    def select_action(self, context: torch.Tensor) -> int:
+        """Select an action based on given context.
+
+        Selecting action with highest predicted reward computed through
+        betas sampled from posterior.
+
+        Args:
+            context (torch.Tensor): The context vector to select action for.
+
+        Returns:
+            int: The action to take.
+        """
+        self.t += 1
+        if self.t < self.n_actions * self.init_pulls:
+            return torch.tensor(
+                self.t % self.n_actions, device=self.device, dtype=torch.int
+            )
+        var = torch.tensor(
+            [self.b[i] * invgamma.rvs(self.a[i]) for i in range(self.n_actions)],
+            device=self.device,
+            dtype=torch.float,
+        )
+        try:
+            beta = (
+                torch.stack(
+                    [
+                        torch.distributions.MultivariateNormal(
+                            self.mu[i], var[i] * self.cov[i]
+                        ).sample()
+                        for i in range(self.n_actions)
+                    ]
+                )
+                .to(self.device)
+                .to(torch.float)
+            )
+        except RuntimeError as e:  # noqa F841
```

Comment:
Seems very fishy to me. Why would you except a generic RuntimeError? Maybe it would be better to initiate default variables to pass instead of `self.mu[i], var[i] * self[i]`?

Reply:
Yeah, I'll change this to a more specific error